pkg/services/sqlstore/team.go

```diff
@@ -189,11 +181,12 @@ func (ss *SQLStore) SearchTeams(ctx context.Context, query *models.SearchTeamsQu
 			params = append(params, user)
 		}
 
-		if query.UserIdFilter == models.FilterIgnoreUser {
-			sql.WriteString(getTeamSelectSQLBase(filteredUsers))
-		} else {
-			sql.WriteString(getTeamSelectWithPermissionsSQLBase(filteredUsers))
+		sql.WriteString(getTeamSelectSQLBase(filteredUsers))
+		if query.UserIdFilter != models.FilterIgnoreUser {
 			params = append(params, query.UserIdFilter)
+		} else {
+			// No filtering applied => see permission for oneself
+			params = append(params, query.SignedInUser.UserId)
```

Comment:
Change in behavior, the signed in user will get his own permission. Not sure if we really want this.